src/test/java/eu/eutampieri/catacombs/tests/CollisionBoxTests.java

```diff
@@ -0,0 +1,50 @@
+package eu.eutampieri.catacombs.tests;
+
+import eu.eutampieri.catacombs.model.CollisionBox;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class CollisionBoxTests {
+
+    private static final CollisionBox MY_BOX = new CollisionBox(5, 10, 5, 5);
+
+    @Test
+    void testOverlapsWithSelf() {
+        assertTrue(MY_BOX.overlaps(MY_BOX));
+    }
+
+    @Test
+    void testDoesntOverlapWithNonOverlappingBox() {
+        final CollisionBox nonOverlapping = new CollisionBox(MY_BOX);
+        nonOverlapping.move(MY_BOX.getWidth() * 2, MY_BOX.getHeight() * 2);
+        assertFalse(MY_BOX.overlaps(nonOverlapping));
+    }
+
+    @Test
+    void testOverlapsWithOverlappingBox() {
+        final CollisionBox overlapping = new CollisionBox(MY_BOX);
+        overlapping.move(MY_BOX.getWidth() / 2, MY_BOX.getHeight() / 2);
+        assertTrue(MY_BOX.overlaps(overlapping));
+    }
+
+    @Test
+    void testDoesntOverlapWithTouchingBox() {
```

Comment:
According to @jackmounts this is the expected behavior

Reply:
It makes sense, if the area of overlap is 0, you can say there is no overlap, this literal edge case shouldn't really matter in the end anyway.